src/eval.py

```
@@ -0,0 +1,71 @@
+    manifest = {
+        "model_checkpoint_hash": model_hash,
+        "eval_dataset": eval_data_hash,
+        "eval_loss": loss.item(),
+        "perplexity": perplexity,
+    }
+    manifest["eval_manifest_hash"] = hash_dict(manifest)
+
+    with open("eval_manifest.json", "w") as f:
+        json.dump(manifest, f, indent=2)
+
+    print(f"\n ~> Manifest saved to eval_manifest.json")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)